### PR TITLE
182031 limitation document

### PR DESCRIPTION
**Pull request by aMasakiTakahashi**
https://acsmine.tok.access-company.com/redmine/issues/182031

### CODE REVIEW

File: doc_src/gear_developers/limitation.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Limitation
+
+There are certain limitations in Antiktyera.
```
Comment (irisTa56) [inline doc_src/gear_developers/limitation.md]:
```suggestion
There are certain limitations in Antikythera.
```

File: doc_src/gear_developers/limitation.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+We introduce some main limitations. But all limitations are not described here. Please also read other parts of the document when you implement gear.
+
+* There is an upper limit of the execution time of web/g2g request
```
Comment (irisTa56) [inline doc_src/gear_developers/limitation.md]:
```suggestion
* There is an upper limit of the execution time of web/g2g request.
```

File: doc_src/gear_developers/limitation.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  * Please refer [here](https://hexdocs.pm/antikythera/development_environment.html#environment-variables-to-tweak-behavior-of-antikythera)
+* There is an upper limit of the execution time of `AsyncJob`.
+  * `AsyncJob` has other limitations too. Please refer [here](https://hexdocs.pm/antikythera/Antikythera.AsyncJob.html#module-registering-jobs)
+* There is an upper limit of heap size which a process can use.
+  * Please refer [here](https://hexdocs.pm/antikythera/development_environment.html#environment-variables-to-tweak-behavior-of-antikythera)
+  * If the size of the heap used by process exceeds the upper limitation, an error log is written.( ex: `xxx killed`)
```
Comment (irisTa56) [inline doc_src/gear_developers/limitation.md]: (just my preference)

```suggestion
  * If the size of the heap used by process exceeds the upper limitation, an error log is written (e.g. `xxx killed`).
```

File: doc_src/gear_developers/limitation.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Limitation
+
+There are certain limitations in Antiktyera.
+We introduce some main limitations. But all limitations are not described here. Please also read other parts of the document when you implement gear.
```
Comment (irisTa56) [inline doc_src/gear_developers/limitation.md]: Is my understanding correct?

```suggestion
We introduce some major limitations here. But this does not cover all the limitations of Antikythera. Please also read other parts of the documentation when you implement gear.
```

File: doc_src/gear_developers/limitation.md (hunk 3 above)
Comment (aDaichiOta) [inline doc_src/gear_developers/limitation.md]: **I think** nouns are better than sentences.

---

* Execution time of web/g2g request
    * Please refer here
* Execution time of AsyncJob
    * AsyncJob has other limitations too. Please refer here
* Heap size which a process can use
    * Please refer here
    * If the size of the heap used by process exceeds the upper limitation, an error log is written.( ex: xxx killed)

File: doc_src/gear_developers/limitation.md (hunk 2 above)
Comment (aMasakiTakahashi) [reply doc_src/gear_developers/limitation.md]: I agree on it.

File: doc_src/gear_developers/limitation.md (hunk 4 above)
Comment (aMasakiTakahashi) [reply doc_src/gear_developers/limitation.md]: Yes, it is.

File: doc_src/gear_developers/limitation.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+There are certain limitations in Antikythera.
+We introduce some major limitations here. But this does not cover all the limitations of Antikythera. Please also read other parts of the documentation when you implement gear.
+
+* There is an upper limit of the execution time of web/g2g request.
+  * Please refer [here](https://hexdocs.pm/antikythera/development_environment.html#environment-variables-to-tweak-behavior-of-antikythera)
+* There is an upper limit of the execution time of `AsyncJob`.
+  * `AsyncJob` has other limitations too. Please refer [here](https://hexdocs.pm/antikythera/Antikythera.AsyncJob.html#module-registering-jobs)
+* There is an upper limit of heap size which a process can use.
+  * Please refer [here](https://hexdocs.pm/antikythera/development_environment.html#environment-variables-to-tweak-behavior-of-antikythera)
+  * If the size of the heap used by process exceeds the upper limitation, an error log is written.( ex: `xxx killed`)
```
Comment (aDaichiOta) [inline doc_src/gear_developers/limitation.md]: How about the following sentence? I think it's clearer. (Please point out if the grammar is unnatural.)
> When a process uses more memory than the limit, Antikythera kills the process

File: doc_src/gear_developers/limitation.md (hunk 1 above)
Comment (aMasakiTakahashi) [reply doc_src/gear_developers/limitation.md]: I fixed in fa2d1e2e1b7a15428bbdf2bb4383356b46f4f95c

File: doc_src/gear_developers/limitation.md (hunk 2 above)
Comment (aMasakiTakahashi) [reply doc_src/gear_developers/limitation.md]: I fixed in 2b732f861c322c0128675ac8847d6e2707104592

File: doc_src/gear_developers/limitation.md (hunk 3 above)
Comment (aMasakiTakahashi) [reply doc_src/gear_developers/limitation.md]: I fixed in 7c54dd4827c67c08f0279a598dc29811ec6a0f69

File: doc_src/gear_developers/limitation.md (hunk 4 above)
Comment (aMasakiTakahashi) [reply doc_src/gear_developers/limitation.md]: I fixed in 5611bc2eaeb9489572dfe9b50ac45127f5d06969

File: doc_src/gear_developers/limitation.md (hunk 3 above)
Comment (aMasakiTakahashi) [reply doc_src/gear_developers/limitation.md]: I think so, too.
I fixed in 7c54dd4827c67c08f0279a598dc29811ec6a0f69

File: doc_src/gear_developers/limitation.md (hunk 5 above)
Comment (aMasakiTakahashi) [reply doc_src/gear_developers/limitation.md]: Thank you for your suggestion.
I fixed in db5eeed34b3dc6a590c37ea9fda16adc5af92943

File: doc_src/gear_developers/limitations.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# Limitations
+
+There are certain limitations in Antikythera.
+We introduce some major limitations here. But this does not cover all the limitations of Antikythera. Please also read other parts of the documentation when you implement gear.
+
+* Execution time of web/g2g request
+  * Please refer [here](https://hexdocs.pm/antikythera/development_environment.html#environment-variables-to-tweak-behavior-of-antikythera).
```
Comment (irisTa56) [inline doc_src/gear_developers/limitations.md]: I felt difficult to find further information in the referenced page.

How about the following?

```markdown
- Execution time of web/g2g request
    - Set via `GEAR_ACTION_TIMEOUT`. Please refer [here](https://hexdocs.pm/antikythera/development_environment.html#environment-variables-to-tweak-behavior-of-antikythera).
- Execution time of AsyncJob
    - Set via `max_duration`. Please refer [here](https://hexdocs.pm/antikythera/Antikythera.AsyncJob.html#module-registering-jobs).
    - AsyncJob has other limitations too.
- Heap size which a process can use
    - Set via `GEAR_PROCESS_MAX_HEAP_SIZE`. Please refer [here](https://hexdocs.pm/antikythera/development_environment.html#environment-variables-to-tweak-behavior-of-antikythera).
```

Comment (aMasakiTakahashi) [reply doc_src/gear_developers/limitations.md]: I think it is better.
I fixed in 7bd85ef75f90e21a65472fbd64b07ee6921ad484

**Comment by irisTa56** [comment]
LGTM!

**Comment by aDaichiOta** [comment]
LGTM!

File: doc_src/gear_developers/limitations.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
+# Limitations
+
+There are certain limitations in Antikythera.
+We list some major limitations here. But this does not cover all the limitations of Antikythera. Please also read other parts of the documentation when you implement gear.
+
+- Execution time of web/g2g request
+    - Set via `GEAR_ACTION_TIMEOUT`. Please refer [here](https://hexdocs.pm/antikythera/development_environment.html#environment-variables-to-tweak-behavior-of-antikythera).
+- Execution time of AsyncJob
```
Comment (ttokit) [inline doc_src/gear_developers/limitations.md]: According to [this page](https://github.com/access-company/antikythera/blob/master/doc_src/gear_developers/async_job.md), in the English context, it is "async job" and the module name is "AsyncJob".
If you want to emphasize aspects of the module, I think you should use single quotes ( `AsyncJob`).
```suggestion
- Execution time of async job
```

**Comment by ttokit** [comment]
Just one comment. LGTM!

File: doc_src/gear_developers/limitations.md (hunk 7 above)
Comment (aMasakiTakahashi) [reply doc_src/gear_developers/limitations.md]: I should have used "async job".
I fixed in c919a54f719be8f28707e2a9b00fb8c9af816cd7 .

**Pull request merged**